homeassistant/components/point/config_flow.py

```diff
@@ -0,0 +1,197 @@
+        CLIENT_ID: client_id,
+        CLIENT_SECRET: client_secret,
+    }
+
+
+@config_entries.HANDLERS.register('point')
+class PointFlowHandler(config_entries.ConfigFlow):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNETION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize flow."""
+        self.flow_impl = None
+
+    async def async_step_import(self, user_input=None):
+        """Handle external yaml configuration."""
+        if self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason='already_setup')
+
+        self.flow_impl = DOMAIN
+
+        return await self.async_step_auth()
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow start."""
+        flows = self.hass.data.get(DATA_FLOW_IMPL, {})
+
+        if self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason='already_setup')
+
+        if not flows:
+            _LOGGER.debug("no flows")
+            return self.async_abort(reason='no_flows')
+
+        if len(flows) == 1:
+            self.flow_impl = list(flows)[0]
+            return await self.async_step_auth()
+
+        if user_input is not None:
+            self.flow_impl = user_input['flow_impl']
+            return await self.async_step_auth()
+
+        return self.async_show_form(
+            step_id='user',
+            data_schema=vol.Schema({
+                vol.Required('flow_impl'):
+                vol.In(list(flows))
+            }))
+
+    async def async_step_auth(self, user_input=None):
+        """Create an entry for auth."""
+        if self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason='external_setup')
+
+        errors = {}
+
+        if user_input is not None:
+            errors['base'] = 'follow_link'
+
+        try:
+            with async_timeout.timeout(10):
+                url = await self._get_authorization_url()
+        except asyncio.TimeoutError:
+            return self.async_abort(reason='authorize_url_timeout')
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected error generating auth url")
+            return self.async_abort(reason='authorize_url_fail')
+
+        return self.async_show_form(
+            step_id='auth',
+            description_placeholders={'authorization_url': url},
+            errors=errors,
+        )
+
+    async def _get_authorization_url(self):
+        """Create Minut Point session and get authorization url."""
+        from pypoint import PointSession
+        flow = self.hass.data[DATA_FLOW_IMPL][self.flow_impl]
+        client_id = flow[CLIENT_ID]
+        client_secret = flow[CLIENT_SECRET]
+        point_session = PointSession(
+            client_id, client_secret=client_secret)
+
+        self.hass.http.register_view(MinutAuthCallbackView())
+
+        return point_session.get_authorization_url
```

Comment:
is this supposed to be not called?

Reply:
it is called, but it should probably be made async (in some way)

Reply:
@balloob, do you want me fix this or is it ok?

Reply:
It is a property that returns the `authoization_url`.

Reply:
As long as it doesn't do I/O it's fine to keep as a property.

Reply:
Ouch, it is doing IO..

Will update that..